test/index.test.js

```diff
@@ -27,3 +31,27 @@ it('creates a new cancel token', () => {
     expect(token).not.toBe(undefined)
     expect(token.token).not.toBe(undefined)
 })
+
+it('intercepting a cancellation error', () => {
```

Comment:
fyi jest tests should be phrased like english sentences starting with *it*. E.g. *it intercepts a cancellation error*. That makes the outputs human readable. ref https://jestjs.io/docs/api#testname-fn-timeout

Reply:
Oh right thanks. Will fix this.

And what about the test content? Are you fine with it?

Reply:
Yes, that looks fine

Reply:
If I may be nitpicky then it would be better if the tests were moved to the individual test suites for the modules